### ISSUE TITLE
Python docs have wrong links to api

### ISSUE THREAD

**Issue by tfardet** [igraph/igraph.org#48]
The [main page](https://igraph.org/python/) of the Python docs has outdated links given the current organization of the docs: it links to [``/python/doc/igraph.Graph-class.html``](https://github.com/igraph/igraph.org/blob/master/python/index.html#L257) and [``/python/doc/igraph.GraphBase-class.html``](https://github.com/igraph/igraph.org/blob/master/python/index.html#L258) while the correct path uses ``python/doc/api/`` (example for [``GraphBase``](https://igraph.org/python/doc/api/igraph._igraph.GraphBase.html)).

Either the docs structure should be reverted (remove api/) or these links should be updated.

**Comment by tristanlatr** [comment]
Hum, this issue comes from the fact the `python-igraph` now uses `pydoctor` instead of `epydoc` to generate it's API documentation. And the naming convention for the pages is not the same as in `epydoc`. 

Probably, the best way to solve this issue is to do a redirection (`xx-class.html` -> `xx.html`) from the web server. 

Not sure how to do this from Github pages, thought.

**Comment by tfardet** [comment]
Not sure what you mean: as far as I can tell, the pages have always been named ``xx-class.html``, the novelty is the ``/api/`` addition after ``/python/doc`` in the URL

**Comment by tristanlatr** [comment]
I see. 

There are two novelties, then. 

As you can see the link the `GraphBase` is `https://igraph.org/python/doc/api/igraph._igraph.GraphBase.html`

**Comment by tfardet** [comment]
Haha, you're right, I missed that one ^^"
Good thing there are two of us!

**Comment by ntamas** [comment]
Thanks for the heads up; we are currently working on versioned documentation pages (see https://github.com/igraph/igraph.org/pull/46 ) and I will fix the broken links when this PR is merged.

As for breaking existing links, I'm afraid there's not much I can do about them; we are using Github Pages, which in turn uses Jekyll, which in turn uses static HTML pages, so I think I cannot simply set up a redirect from pages-matching-some-regex to some other destination. I will nevertheless look into whether there is a Jekyll plugin that I could use to ease the transition.

**Comment by ntamas** [comment]
_Edit_: it looks like the `jekyll-redirect-from` plugin might be able to set up the redirects from the old links to the new ones.

**Comment by tfardet** [comment]
> _Edit_: it looks like the `jekyll-redirect-from` plugin might be able to set up the redirects from the old links to the new ones.

OK, sounds nice!

Just to confirm, the new structure of the links is supposed to be the new standard (i.e. it will remain that way) right?
Because I am referencing links to igraph documentation in [NNGT](https://nngt.readthedocs.io/) so I'd like to update them tothe preferred link format is.

**Comment by ntamas** [comment]
Yes, we aim to keep the link structure for the foreseeable future. The Python API doc (generated from `pydoctor`) will live under `/python/api/<version>`, where `<version>` will be either a released `python-igraph` version, or `master` for the head of the master branch and `develop` for the head of the development branch. (I'm not sure how often the `master` / `development` pages will be updated, we still have to figure out how to fit this into our workflow). You will also be able to use `latest` in place of the version number, which will lead you to the latest released version (0.9.6 at the time of writing). The old links that are in effect on igraph.org _now_ will redirect to `latest`.

**Comment by tfardet** [comment]
ah OK, so I should rather wait for #46 to be merged!

**Comment by ntamas** [comment]
#46 is now merged and deployed to igraph.org; let me know if you find any issues or broken links.

**Comment by tfardet** [comment]
Sorry @ntamas, one little detail: the redirection does not seem to keep the anchors.
This means that for instance: [https://igraph.org/python/api/latest/igraph._igraph.GraphBase.html#get_all_shortest_paths](https://igraph.org/python/api/latest/igraph._igraph.GraphBase.html#get_all_shortest_paths) redirects to https://igraph.org/python/api/0.9.6/igraph._igraph.GraphBase.html...

Since preserving the anchors using ``jekyll-redirect-from`` might be [more than you want to do](https://github.com/jekyll/jekyll-redirect-from/issues/219), would it be possible to have an actual ``lastest`` or ``stable`` entry that would be equivalent right now to 0.9.6 but with a redirect so there is a way to link to igraph without having to specify a version?

**Comment by ntamas** [comment]
It has been brought to my attention that Github Pages seems to support symlinks so I will try to update our pipeline to make `latest` a symlink to the most recent version. I hope that it will solve the problem with the anchors, but I can do it only tomorrow or the day after.
